spec/helpers/govuk_rails_compatibile_link_helper_spec.rb

```diff
@@ -0,0 +1,325 @@
+require 'spec_helper'
+
+RSpec.describe(GovukRailsCompatibileLinkHelper, type: 'helper') do
+  include ActionView::Helpers::UrlHelper
+  include ActionView::Context
+  include GovukRailsCompatibileLinkHelper
+
+  let(:text) { 'Menu' }
+  let(:url) { '/stuff/menu/' }
+  let(:args) { [text, url] }
+  let(:kwargs) { {} }
+
+  describe '#govuk_link_classes' do
+    subject { govuk_link_classes(*args) }
+
+    describe "by default" do
+      let(:args) { [] }
+
+      specify "contains only 'govuk-link'" do
+        expect(subject).to eql(%w(govuk-link))
+      end
+    end
+
+    {
+      no_visited_state: 'govuk-link--no-visited-state',
+      muted:            'govuk-link--muted',
+      text_colour:      'govuk-link--text-colour',
+      inverse:          'govuk-link--inverse',
+      no_underline:     'govuk-link--no-underline',
+    }.each do |style, css_class|
+      describe "generating a #{style}-style link with '#{style}: true'" do
+        let(:args) { [style] }
+
+        specify %(contains 'govuk-link' and '#{css_class}') do
+          expect(subject).to match_array(['govuk-link', css_class])
+        end
+      end
+    end
+  end
+
+  describe '#govuk_button_classes' do
+    subject { govuk_button_classes(*args) }
+
+    describe "by default" do
+      let(:args) { [] }
+
+      specify "contains only 'govuk-button'" do
+        expect(subject).to eql(%w(govuk-button))
+      end
+    end
+
+    {
+      secondary: 'govuk-button--secondary',
+      warning:   'govuk-button--warning',
+      disabled:  'govuk-button--disabled',
+      inverse:   'govuk-button--inverse',
+    }.each do |style, css_class|
+      describe "generating a #{style}-style button with '#{style}: true'" do
+        let(:args) { [style] }
+
+        specify %(contains 'govuk-button' and '#{css_class}') do
+          expect(subject).to match_array(['govuk-button', css_class])
+        end
+      end
+    end
+  end
+
+  describe "#govuk_link_to" do
+    let(:link_text) { 'Onwards!' }
+    let(:link_url) { '/some/link' }
+    let(:link_params) { { controller: :some_controller, action: :some_action } }
```

Comment:
Confused by this - thought you’d deprecated this method, and now always require a path to be set instead of setting a controller and action?

Reply:
This covers the old helpers which are still present (in `GovukRailsCompatibileLinkHelper`) for teams working on codebases that won't support the new style.